src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/resource_allocation_manager_impl.cc

```diff
@@ -213,10 +219,133 @@ void ResourceAllocationManagerImpl::RemoveAppsSubscriptions(const Apps& apps) {
+  msg_params[message_params::kFreeModules] = free_modules;
+}
+
+smart_objects::SmartObjectSPtr
+ResourceAllocationManagerImpl::CreateOnRCStatusNotificationToMobile(
+    const application_manager::ApplicationSharedPtr app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using application_manager::MessageHelper;
+  auto msg_to_mobile = MessageHelper::CreateNotification(
+      mobile_apis::FunctionID::OnRCStatusID, app->app_id());
+  auto& msg_params = (*msg_to_mobile)[application_manager::strings::msg_params];
+  if (is_rc_enabled()) {
+    ConstructOnRCStatusNotificationParams(msg_params,
+                                          allocated_resources_,
+                                          all_supported_modules(),
+                                          app->app_id());
+  } else {
+    msg_params[message_params::kAllocatedModules] =
+        smart_objects::SmartObject(smart_objects::SmartType_Array);
+    msg_params[message_params::kFreeModules] =
+        smart_objects::SmartObject(smart_objects::SmartType_Array);
+  }
+  return msg_to_mobile;
+}
+
+smart_objects::SmartObjectSPtr
+ResourceAllocationManagerImpl::CreateOnRCStatusNotificationToHmi(
+    const application_manager::ApplicationSharedPtr app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  using application_manager::MessageHelper;
+  auto msg_to_hmi =
+      MessageHelper::CreateHMINotification(hmi_apis::FunctionID::RC_OnRCStatus);
+  auto& msg_params = (*msg_to_hmi)[application_manager::strings::msg_params];
+  ConstructOnRCStatusNotificationParams(
+      msg_params, allocated_resources_, all_supported_modules(), app->app_id());
+  msg_params[application_manager::strings::app_id] = app->hmi_app_id();
+  return msg_to_hmi;
+}
+
 void ResourceAllocationManagerImpl::SetResourceAquired(
     const std::string& module_type, const uint32_t app_id) {
   LOG4CXX_AUTO_TRACE(logger_);
   allocated_resources_[module_type] = app_id;
+  SendOnRCStatusNotifications(
+      NotificationTrigger::MODULE_ALLOCATION,
+      utils::SharedPtr<application_manager::Application>());
+}
+
+void ResourceAllocationManagerImpl::SendOnRCStatusNotifications(
+    NotificationTrigger::eType event,
+    application_manager::ApplicationSharedPtr application) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  smart_objects::SmartObjectSPtr msg_to_mobile;
+  smart_objects::SmartObjectSPtr msg_to_hmi;
+  if (NotificationTrigger::APP_REGISTRATION == event) {
+    DCHECK(application);
+    msg_to_mobile = CreateOnRCStatusNotificationToMobile(application);
+    (*msg_to_mobile)[application_manager::strings::msg_params]
+                    [message_params::kAllowed] = is_rc_enabled();
+    rpc_service_.SendMessageToMobile(msg_to_mobile);
+  } else {
+    auto rc_apps = RCRPCPlugin::GetRCApplications(app_mngr_);
+    for (const auto& rc_app : rc_apps) {
+      msg_to_mobile = CreateOnRCStatusNotificationToMobile(rc_app);
+      if (NotificationTrigger::RC_STATE_CHANGING == event) {
```

Comment:
I believe that the allowed parameter should be included for all reasons including `MODULE_ALLOCATION`

Reply:
@JackLivio As allowed parameter is **not mandatory**, so conditions for it sending must be specified, please take a look at https://github.com/smartdevicelink/sdl_requirements/issues/84

Reply:
The requirement you linked to states 
> SDL does not sends allowed parameter by module allocation/deallocation.

But that behavior is not defined anywhere in the actual proposal.

Reply:
@JackLivio Yes, I agree with you.
@yang1070 What is your opinion regarding this case?

Reply:
@JackLivio The intention of the proposal is to let the app know the event change on HMI side that the user enables or disables the RC.  the newly added parameter is specially for this change event. The proposal itself does not clearly define whether the parameter shall be include or not in case of resource allocation/dis-allocation. I think that implies a not including because there is no change on it. 
However, always include the parameter is also OK for me, as that makes the SDL logic simpler.

Reply:
@yang1070 Thank you for the response, I think I would prefer to keep the logic simpler. It was also causing some confusion on the proxy side when our team was conducting the review.

Reply:
@JackLivio shouldn't we make this parameter mandatory in that case?